### PR TITLE
Feature/one column template

### PR DESCRIPTION
**Pull request by KarthikTG47**

### CODE REVIEW

File: bootcamp-demo-project/ui.apps/src/main/content/jcr_root/apps/bootcamp-demo-project/components/content/footer/.content.xml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<jcr:root xmlns:cq="http://www.day.com/jcr/cq/1.0" xmlns:jcr="http://www.jcp.org/jcr/1.0"
+    jcr:primaryType="cq:Component"
+    jcr:title="Footer Component"
+    componentGroup="bootcamp-demo-project"/>
```
Comment (RaghavendraNagaraj) [inline bootcamp-demo-project/ui.apps/src/main/content/jcr_root/apps/bootcamp-demo-project/components/content/footer/.content.xml]: Add footer to .hidden group. Authors should not include this

File: bootcamp-demo-project/ui.apps/src/main/content/jcr_root/apps/bootcamp-demo-project/components/content/header/.content.xml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<jcr:root xmlns:cq="http://www.day.com/jcr/cq/1.0" xmlns:jcr="http://www.jcp.org/jcr/1.0"
+    jcr:primaryType="cq:Component"
+    jcr:title="Header Component"
+    componentGroup="bootcamp-demo-project"/>
```
Comment (RaghavendraNagaraj) [inline bootcamp-demo-project/ui.apps/src/main/content/jcr_root/apps/bootcamp-demo-project/components/content/header/.content.xml]: Add header to .hidden group

File: bootcamp-demo-project/ui.apps/src/main/content/jcr_root/apps/bootcamp-demo-project/templates/one-column-template/.content.xml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<jcr:root xmlns:sling="http://sling.apache.org/jcr/sling/1.0" xmlns:cq="http://www.day.com/jcr/cq/1.0" xmlns:jcr="http://www.jcp.org/jcr/1.0"
+    jcr:primaryType="cq:Template"
+    jcr:title="one-column-template"
+    allowedPaths="[/content(/.*)?]"
+    ranking="{Long}100">
+    <jcr:content
+        jcr:primaryType="cq:PageContent"
+        sling:resourceType="bootcamp-demo-project/components/structure/one-column"/>
+        <header
```
Comment (RaghavendraNagaraj) [inline bootcamp-demo-project/ui.apps/src/main/content/jcr_root/apps/bootcamp-demo-project/templates/one-column-template/.content.xml]: Why is this required?

File: bootcamp-demo-project/ui.apps/src/main/content/jcr_root/apps/bootcamp-demo-project/components/content/footer/.content.xml (hunk 1 above)
Comment (KarthikTG47) [reply bootcamp-demo-project/ui.apps/src/main/content/jcr_root/apps/bootcamp-demo-project/components/content/footer/.content.xml]: done.

File: bootcamp-demo-project/ui.apps/src/main/content/jcr_root/apps/bootcamp-demo-project/templates/one-column-template/.content.xml (hunk 3 above)
Comment (KarthikTG47) [reply bootcamp-demo-project/ui.apps/src/main/content/jcr_root/apps/bootcamp-demo-project/templates/one-column-template/.content.xml]: because once I create the template, in order to use it in page creation, I need a component to which I link my template. If that is not there, I won't be able to create the page, it will not show the dialog box where we configure the page title.

File: bootcamp-demo-project/ui.apps/src/main/content/jcr_root/apps/bootcamp-demo-project/components/content/header/.content.xml (hunk 2 above)
Comment (KarthikTG47) [reply bootcamp-demo-project/ui.apps/src/main/content/jcr_root/apps/bootcamp-demo-project/components/content/header/.content.xml]: done, sir.
